### PR TITLE
MAINT: simplify test with None and NullObject

### PR DESCRIPTION
**Pull request by pubpub-zz**
This will replace test "is (not) None" in order to check also for NullObject

to be reviewed after #2811

### CODE REVIEW

**Comment by pubpub-zz** [comment]
@stefan6419846 
I do not understand what's wrong 😫

any idea ?

**Comment by stefan6419846** [comment]
Apparently the coverage is being recorded, but somehow the corresponding artifact cannot be found for upload:

> No files were found with the provided path: .coverage.*. No artifacts will be uploaded.

We might have to check which files are actually there using `ls -la` before uploading the coverage data to detect possible issues.

**Comment by stefan6419846** [comment]
Should be rebased after #2820 has been merged.

File: pypdf/generic/_base.py [hunk 1]
```diff
@@ -214,6 +214,16 @@ def __repr__(self) -> str:
         return "NullObject"
 
 
+def is_null_or_none(x: Any) -> bool:
+    """
+    Returns:
+        returns true if x is None or NullObject
```
Comment (stefan6419846) [inline pypdf/generic/_base.py:220]:
```suggestion
        True if x is None or NullObject.
```

File: pypdf/generic/_base.py [hunk 2]
```diff
@@ -214,6 +214,16 @@ def __repr__(self) -> str:
         return "NullObject"
 
 
+def is_null_or_none(x: Any) -> bool:
```
Comment (stefan6419846) [inline pypdf/generic/_base.py:217]: I propose adding a test for this as well to check the basic behavior.

Comment (pubpub-zz) [reply pypdf/generic/_base.py:217]: added

**Review by stefan6419846 — APPROVED** [review]

**Comment by pubpub-zz** [comment]
I think we can now issue 5.0.0

**Comment by stefan6419846** [comment]
Seems like you already proposed it in the meantime.

**Pull request merged**